locale.go

```diff
@@ -109,9 +112,12 @@ func (service *LocalesService) Upsert(ctx context.Context, spaceID string, local
 	var path string
 	var method string
 
-	if locale.Sys != nil && locale.Sys.CreatedAt != "" {
+	if locale.Sys != nil && locale.Sys.CreatedAt != ""  {
 		path = fmt.Sprintf("/spaces/%s/locales/%s", spaceID, locale.Sys.ID)
```

Comment:
I think `locale` always needs environments. https://www.contentful.com/developers/docs/references/content-delivery-api/#/reference/locales

So, Could you change it? And you can use `service.c.Environment` instead of adding new field.

```suggestion
		path = fmt.Sprintf("/spaces/%s/environments/%s/locales/%s", spaceID, service.c.Environment, locale.Sys.ID)
```

Reply:
that looks like the Content Delivery API, are we using the CMA instead? https://www.contentful.com/developers/docs/references/content-management-api/#/reference/locales/locale-collection/create-a-locale/console